crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -597,13 +604,19 @@ impl PgManager {
             }
 
             if let Some(owner) = filter.owner {
+                // Default to Address owner so we can leverage the index
+                let owner_type = owner_type.unwrap_or(OwnerType::Address);
```

Comment:
There is an objects_owner btree index on (owner_type, owner_id), and the upper bound on cost is much lower if we can discern whether owner_type==1 (Address) or ==2 (ObjectOwner). iirc most cases of ownership fall under AddressOwner, ObjectOwner is specific for dynamic fields, and shared objects and immutable objects don't have owners (?) so we can use owner_type==1 for most cases and owner_type==2 for dynamic field fetches. This arg might be better represented as an enum for just the two owner kinds instead of an Option<OwnerType>

explain select object_id from objects where owner_id='\x8110684da65283bc2cbddb8af0a1300e1b4e74031ffa5d57d639295b9d1b6a10' and owner_type between 1 and 2;
                                               QUERY PLAN                
--------------------------------------------------------------------------------------------------------
 Index Scan using objects_owner on objects  (cost=0.56..1053712.99 rows=648 width=33)
   Index Cond: (owner_id = '\x8110684da65283bc2cbddb8af0a1300e1b4e74031ffa5d57d639295b9d1b6a10'::bytea)
(2 rows)

defaultdb=> explain select object_id from objects where owner_id='\x8110684da65283bc2cbddb8af0a1300e1b4e74031ffa5d57d639295b9d1b6a10' and owner_type=1;
                                                          QUERY PLAN     
-------------------------------------------------------------------------------------------------------------------------------
 Index Scan using objects_owner on objects  (cost=0.56..1140.89 rows=319 width=33)
   Index Cond: ((owner_type = 1) AND (owner_id = '\x8110684da65283bc2cbddb8af0a1300e1b4e74031ffa5d57d639295b9d1b6a10'::bytea))
(2 rows)

defaultdb=> explain select object_id from objects where owner_id='\x8110684da65283bc2cbddb8af0a1300e1b4e74031ffa5d57d639295b9d1b6a10' and owner_type=2;
                                                          QUERY PLAN     
-------------------------------------------------------------------------------------------------------------------------------
 Index Scan using objects_owner on objects  (cost=0.56..1179.91 rows=330 width=33)
   Index Cond: ((owner_type = 2) AND (owner_id = '\x8110684da65283bc2cbddb8af0a1300e1b4e74031ffa5d57d639295b9d1b6a10'::bytea))

Reply:
> This arg might be better represented as an enum for just the two owner kinds instead of an Option

Right, this is what I was referring to.  It's confusing that `None` is the same as `Some(AddressOwner)` especially because we have kinds of object that have no owner (like shared and immutable objects, which you mentioned).

I agree that it makes good sense to always require an owner type filter, to leverage the index -- my issue is with the conflation of `None` and `Some(AddressOwner)`.